### PR TITLE
Feature/attribution

### PR DESCRIPTION
**Pull request by sbellity**

### CODE REVIEW

File: src/client/current-config.coffee [hunk 1]
```diff
@@ -95,7 +112,45 @@ class CurrentConfig
...
+    rawValue = cookies.get(cookieName)
+    if rawValue
+      try
+        ident = JSON.parse(Base64.decode(rawValue))
+      catch err
+        ident = {}
+    else
+      ident = {}
+
+    ident = _.defaults(ident || {}, {
```
Comment (unity) [inline src/client/current-config.coffee]: use assign ?

File: src/client/current-config.coffee [hunk 2]
```diff
@@ -14,6 +26,8 @@ getRemoteUrl = (config)->
   url += "&debug_remote=true"   if config.debugRemote
   url += "&access_token=#{config.accessToken}" if config.accessToken?
   url += "&user_hash=#{config.userHash}" if config.userHash != undefined
+  url += "&_bid=#{identifiers.bid}" if identifiers?.bid
+  url += "&_sid=#{identifiers.sid}" if identifiers?.sid
```
Comment (oliger) [inline src/client/current-config.coffee]: I thought we decided to rename _bid to device_id. I do not think that prefixing those params with underscore is needed.

File: src/client/current-config.coffee [hunk 3]
```diff
@@ -52,6 +66,17 @@ checkConfig = (config)->
   promise.then(null, throwErr)
   promise
 
+extractDomainFromUrl = (url)->
+  return unless url && url.length
+  u = url.match(/^(https?:\/\/)?([\da-z\.-]+)([\/\w \.-]*)*\/?/)
+  u[2] if u
+
+extractUtmTags = ->
```
Comment (oliger) [inline src/client/current-config.coffee]: We have `qs`

File: src/client/current-config.coffee (hunk 2 above)
Comment (oliger) [inline src/client/current-config.coffee]: I think that using `Qs` to build the query string will make the code cleaner.

**Comment by oliger** [comment]
Looks ok to me :+1:. Can you tell me if we will rename `browser_id` to something else ?

**Pull request merged**